# Исследование игровой индустрии за период с 2000 по 2013 года

- Автор: Завьялов Иван Викторович
- Дата: 13.06.2025

<h1>Разделы проекта<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><ul class="toc-item"><li><span><a href="#Цели-и-задачи-проекта" data-toc-modified-id="Цели-и-задачи-проекта-0.1"><span class="toc-item-num">0.1&nbsp;&nbsp;</span>Цели и задачи проекта</a></span></li><li><span><a href="#Описание-данных" data-toc-modified-id="Описание-данных-0.2"><span class="toc-item-num">0.2&nbsp;&nbsp;</span>Описание данных</a></span></li><li><span><a href="#Содержимое-проекта" data-toc-modified-id="Содержимое-проекта-0.3"><span class="toc-item-num">0.3&nbsp;&nbsp;</span>Содержимое проекта</a></span></li></ul></li><li><span><a href="#Загрузка-данных-и-знакомство-с-ними" data-toc-modified-id="Загрузка-данных-и-знакомство-с-ними-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Загрузка данных и знакомство с ними</a></span></li><li><span><a href="#Проверка-ошибок-в-данных-и-их-предобработка" data-toc-modified-id="Проверка-ошибок-в-данных-и-их-предобработка-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Проверка ошибок в данных и их предобработка</a></span><ul class="toc-item"><li><span><a href="#Названия,-или-метки,-столбцов-датафрейма" data-toc-modified-id="Названия,-или-метки,-столбцов-датафрейма-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Названия, или метки, столбцов датафрейма</a></span></li><li><span><a href="#Типы-данных" data-toc-modified-id="Типы-данных-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>Типы данных</a></span></li><li><span><a href="#Наличие-пропусков-в-данных" data-toc-modified-id="Наличие-пропусков-в-данных-2.3"><span class="toc-item-num">2.3&nbsp;&nbsp;</span>Наличие пропусков в данных</a></span></li><li><span><a href="#Явные-и-неявные-дубликаты-в-данных" data-toc-modified-id="Явные-и-неявные-дубликаты-в-данных-2.4"><span class="toc-item-num">2.4&nbsp;&nbsp;</span>Явные и неявные дубликаты в данных</a></span></li></ul></li><li><span><a href="#Фильтрация-данных" data-toc-modified-id="Фильтрация-данных-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Фильтрация данных</a></span></li><li><span><a href="#Категоризация-данных" data-toc-modified-id="Категоризация-данных-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Категоризация данных</a></span></li><li><span><a href="#Итоговый-вывод" data-toc-modified-id="Итоговый-вывод-5"><span class="toc-item-num">5&nbsp;&nbsp;</span>Итоговый вывод</a></span></li></ul></div>

### Цели и задачи проекта

<font color='#777778'>Задачами проекта являются: анализ развития игровой индустрии в разных регионах за период с 2000 по 2013 год; выявление самых популярных платформ по количеству выпущенных игр и самых популярных игр по оценкам критиков и пользователей. Целью проекта является популяризация старых игр для привлечения новых пользователей.</font>

### Описание данных

<font color='#777778'>Для исследования используем исторические данные по играм, собранные из окрытых источников. Данные содержатся в файле /datasets/new_games.csv. Они содержат информацию о продажах игр разных жанров и платформ, а также пользовательские и экспертные оценки. </font>

### Содержимое проекта

1. Загрузка и знакомство с данными
2. Проверка ошибок в данных и их предобработка
3. Фильтрация данных
4. Категоризация данных
5. Итоговый вывод по проекту

---

## Загрузка данных и знакомство с ними

- Загрузим библиотеку Pandas и данные датасета `/datasets/new_games.csv`.


In [1]:
import pandas as pd# Пустые ячейки после каждого задания — примерное пространство для работы.
# Вы можете свободно добавлять или удалять ячейки по своему усмотрению в зависимости от логики и объёма работы.

In [2]:
df=pd.read_csv('/datasets/new_games.csv')# Используйте ячейки типа Code для вашего кода

- Познакомимся с данными: выведем первые и последние строки, а также информацию по датасету с помощью  метода `info()`.


In [3]:
df.head()

Name Platform  Year of Release         Genre  NA sales  \
0                Wii Sports      Wii           2006.0        Sports     41.36   
1         Super Mario Bros.      NES           1985.0      Platform     29.08   
2            Mario Kart Wii      Wii           2008.0        Racing     15.68   
3         Wii Sports Resort      Wii           2009.0        Sports     15.61   
4  Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing     11.27   

  EU sales JP sales  Other sales  Critic Score User Score Rating  
0    28.96     3.77         8.45          76.0          8      E  
1     3.58     6.81         0.77           NaN        NaN    NaN  
2    12.76     3.79         3.29          82.0        8.3      E  
3    10.93     3.28         2.95          80.0          8      E  
4     8.89    10.22         1.00           NaN        NaN    NaN

In [4]:
df.tail()

Name Platform  Year of Release       Genre  \
16951  Samurai Warriors: Sanada Maru      PS3           2016.0      Action   
16952               LMA Manager 2007     X360           2006.0      Sports   
16953        Haitaka no Psychedelica      PSV           2016.0   Adventure   
16954               Spirits & Spells      GBA           2003.0    Platform   
16955            Winning Post 8 2016      PSV           2016.0  Simulation   

       NA sales EU sales JP sales  Other sales  Critic Score User Score Rating  
16951      0.00      0.0     0.01          0.0           NaN        NaN    NaN  
16952      0.00     0.01      0.0          0.0           NaN        NaN    NaN  
16953      0.00      0.0     0.01          0.0           NaN        NaN    NaN  
16954      0.01      0.0      0.0          0.0           NaN        NaN    NaN  
16955      0.00      0.0     0.01          0.0           NaN        NaN    NaN

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16956 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16954 non-null  object 
 1   Platform         16956 non-null  object 
 2   Year of Release  16681 non-null  float64
 3   Genre            16954 non-null  object 
 4   NA sales         16956 non-null  float64
 5   EU sales         16956 non-null  object 
 6   JP sales         16956 non-null  object 
 7   Other sales      16956 non-null  float64
 8   Critic Score     8242 non-null   float64
 9   User Score       10152 non-null  object 
 10  Rating           10085 non-null  object 
dtypes: float64(4), object(7)
memory usage: 1.4+ MB


- Для исседования предоставлены данные в объёме 16956 строк и 11 столбцов. Данные соответствуют предоставленному описанию. В 6 сериях содержатся пропуски в данных,больше всего в оценках пользователей и критиков, а также рейтинга игры.Некоторые типы данных в сериях не соответствуют значениям этой серии и их можно поменять для корректности анализа.
- Названия серий отражают содержащиеся в них данные, но формат названия серий нужно поменять на общепринятый стиль snake case.

- Для обработки типов данных нужно сделать следующее: 
1.Серии с названием игры, названием платформы,рейтинга и жанра представлены типом данных object, что соответствует их содержанию, оставляем данный тип. 2.Серия с годом релиза игры представлена вещественным типом данных float64 и его лучше заменить на целочисленный тип int64. 3.Серии с данными о продажах в Европе и Японии представлены типом данных object и его лучше поменять на вещественный тип данных float64. 4.Серии с продажами в Северной Америке, с продажами в других регионах и оценками критиков соответствуют типу данных по своему содержанию и их оставляем, а серию с оценкой пользователей нужно привести к вещественному типу float64.

- Для обработки пропусков в данных: 
1.В сериях с названием игры и жанром пропущено по 2 значения и эти пропуски можно удалить, так как они не повлияют на анализ. 2.В сериях с продажами пропуски нужно будет обработать для более точного анализа.
3.В сериях с оценками и рейтингом пропуски можно будет оставить, но обозначить при категоризации данных, как отдельную категорию.

---

##  Проверка ошибок в данных и их предобработка


###  Названия, или метки, столбцов датафрейма



- Выведем названия всех столбцов и проверим их стиль написания.

In [6]:
display(df.columns.tolist())

['Name',
 'Platform',
 'Year of Release',
 'Genre',
 'NA sales',
 'EU sales',
 'JP sales',
 'Other sales',
 'Critic Score',
 'User Score',
 'Rating']

- Приведём названия столбцов к стилю snake case и нижнему регистру.

In [7]:
df=df.rename(columns={'Name':'name','Platform':'platform','Year of Release':'year_of_release','Genre':'genre','NA sales':'na_sales','EU sales':'eu_sales','JP sales':'jp_sales', 'Other sales':'other_region_sales','Critic Score':'critic_score','User Score':'user_score','Rating':'rating'})

- Выведем на экран обновлённые названия столбцов.

In [8]:
display(df.columns.tolist())

['name',
 'platform',
 'year_of_release',
 'genre',
 'na_sales',
 'eu_sales',
 'jp_sales',
 'other_region_sales',
 'critic_score',
 'user_score',
 'rating']

- Сохраним длину первоначального датасета в переменную для оценки показателей последующей очистки данных.

In [9]:
start_string_count=df.shape[0]

### Типы данных



- В годе релиза данные представлены числовым типом float64,что соответствует их содержанию. 

- Столбцы с продажами в Европе и Японии представлены типом данных object, что связано с наличием строковых занчений 'unknown' в данных. Для приведения их к вещественному типу,заменим строковые значения на пропуски и и приведём тип данных к float64.

- Выведем уникальные значения столбцов с продажами в Европе и Японии.

In [10]:
display(df['eu_sales'].unique())

array(['28.96', '3.58', '12.76', '10.93', '8.89', '2.26', '9.14', '9.18',
       '6.94', '0.63', '10.95', '7.47', '6.18', '8.03', '4.89', '8.49',
       '9.09', '0.4', '3.75', '9.2', '4.46', '2.71', '3.44', '5.14',
       '5.49', '3.9', '5.35', '3.17', '5.09', '4.24', '5.04', '5.86',
       '3.68', '4.19', '5.73', '3.59', '4.51', '2.55', '4.02', '4.37',
       '6.31', '3.45', '2.81', '2.85', '3.49', '0.01', '3.35', '2.04',
       '3.07', '3.87', '3.0', '4.82', '3.64', '2.15', '3.69', '2.65',
       '2.56', '3.11', '3.14', '1.94', '1.95', '2.47', '2.28', '3.42',
       '3.63', '2.36', '1.71', '1.85', '2.79', '1.24', '6.12', '1.53',
       '3.47', '2.24', '5.01', '2.01', '1.72', '2.07', '6.42', '3.86',
       '0.45', '3.48', '1.89', '5.75', '2.17', '1.37', '2.35', '1.18',
       '2.11', '1.88', '2.83', '2.99', '2.89', '3.27', '2.22', '2.14',
       '1.45', '1.75', '1.04', '1.77', '3.02', '2.75', '2.16', '1.9',
       '2.59', '2.2', '4.3', '0.93', '2.53', '2.52', '1.79', '1.3', '2.6',
   

In [11]:
display(df['jp_sales'].unique())

array(['3.77', '6.81', '3.79', '3.28', '10.22', '4.22', '6.5', '2.93',
       '4.7', '0.28', '1.93', '4.13', '7.2', '3.6', '0.24', '2.53',
       '0.98', '0.41', '3.54', '4.16', '6.04', '4.18', '3.84', '0.06',
       '0.47', '5.38', '5.32', '5.65', '1.87', '0.13', '3.12', '0.36',
       '0.11', '4.35', '0.65', '0.07', '0.08', '0.49', '0.3', '2.66',
       '2.69', '0.48', '0.38', '5.33', '1.91', '3.96', '3.1', '1.1',
       '1.2', '0.14', '2.54', '2.14', '0.81', '2.12', '0.44', '3.15',
       '1.25', '0.04', '0.0', '2.47', '2.23', '1.69', '0.01', '3.0',
       '0.02', '4.39', '1.98', '0.1', '3.81', '0.05', '2.49', '1.58',
       '3.14', '2.73', '0.66', '0.22', '3.63', '1.45', '1.31', '2.43',
       '0.7', '0.35', '1.4', '0.6', '2.26', '1.42', '1.28', '1.39',
       '0.87', '0.17', '0.94', '0.19', '0.21', '1.6', '0.16', '1.03',
       '0.25', '2.06', '1.49', '1.29', '0.09', '2.87', '0.03', '0.78',
       '0.83', '2.33', '2.02', '1.36', '1.81', '1.97', '0.91', '0.99',
       '0.95', '2.0'

- Заменим строковые значения на пропуски и приведём к типу float64.

In [12]:
df[['eu_sales','jp_sales']]=df[['eu_sales','jp_sales']].replace('unknown','Nan')

In [13]:
df[['eu_sales','jp_sales']]=df[['eu_sales','jp_sales']].astype('float64')

- В столбце с оценками пользователей тип данных object из-за наличия строкового значения 'tbd'. Заменим его на пропуск и приведём тип данных к float64.

- Выведем уникальные значения с столбце с оценкой пользователей.

In [14]:
display(df['user_score'].unique())

array(['8', nan, '8.3', '8.5', '6.6', '8.4', '8.6', '7.7', '6.3', '7.4',
       '8.2', '9', '7.9', '8.1', '8.7', '7.1', '3.4', '5.3', '4.8', '3.2',
       '8.9', '6.4', '7.8', '7.5', '2.6', '7.2', '9.2', '7', '7.3', '4.3',
       '7.6', '5.7', '5', '9.1', '6.5', 'tbd', '8.8', '6.9', '9.4', '6.8',
       '6.1', '6.7', '5.4', '4', '4.9', '4.5', '9.3', '6.2', '4.2', '6',
       '3.7', '4.1', '5.8', '5.6', '5.5', '4.4', '4.6', '5.9', '3.9',
       '3.1', '2.9', '5.2', '3.3', '4.7', '5.1', '3.5', '2.5', '1.9', '3',
       '2.7', '2.2', '2', '9.5', '2.1', '3.6', '2.8', '1.8', '3.8', '0',
       '1.6', '9.6', '2.4', '1.7', '1.1', '0.3', '1.5', '0.7', '1.2',
       '2.3', '0.5', '1.3', '0.2', '0.6', '1.4', '0.9', '1', '9.7'],
      dtype=object)

- Заменим строковые значения на пропуски и приведём к типу данных float64.

In [15]:
df['user_score']=df['user_score'].replace('tbd','nan')

In [16]:
df['user_score']=df['user_score'].astype('float64')

- Выведем обновлённую информацию о датасете с верными типами данных

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16956 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   name                16954 non-null  object 
 1   platform            16956 non-null  object 
 2   year_of_release     16681 non-null  float64
 3   genre               16954 non-null  object 
 4   na_sales            16956 non-null  float64
 5   eu_sales            16950 non-null  float64
 6   jp_sales            16952 non-null  float64
 7   other_region_sales  16956 non-null  float64
 8   critic_score        8242 non-null   float64
 9   user_score          7688 non-null   float64
 10  rating              10085 non-null  object 
dtypes: float64(7), object(4)
memory usage: 1.4+ MB


###  Наличие пропусков в данных




- Посчитаем количество пропусков в датасете в абсолютных значениях.

In [18]:
df.isna().sum()

name                     2
platform                 0
year_of_release        275
genre                    2
na_sales                 0
eu_sales                 6
jp_sales                 4
other_region_sales       0
critic_score          8714
user_score            9268
rating                6871
dtype: int64

- Посчитаем долю пропусков в процентах % и округлим до сотых.

In [19]:
display(round(df.isna().sum()/len(df)*100,2))

name                   0.01
platform               0.00
year_of_release        1.62
genre                  0.01
na_sales               0.00
eu_sales               0.04
jp_sales               0.02
other_region_sales     0.00
critic_score          51.39
user_score            54.66
rating                40.52
dtype: float64

- В данных присутствуют пропуски: в столбце с годом релиза 275 пропусков; в столбце с названием игры-2 пропуска; в столбце с жанром-2 пропуска;в столбцах с продажами в Европе и Японии по 6 и 4 пропуска соответственно;в оценках пользователей,критиков и присвоенным рейтингом большое количество пропусков. Пропуски в столбце с годом релиза возможно связаны с недостатком информации в открытых источниках и давностью выхода этих игр. ИХ можно заменить на значение-индикатор. Пропуски в названии игры и жанре возможно связаны с технической ошибкой и из-за небольшого их количества их можно удалить. Пропуски в столбцах с количеством продаж в Европе и Японии можно заменить  на среднее значение в зависимости от названия платформы и года выхода игры. Пропуски в оценках пользователей,критиков и рейтингах возможно частично связаны с годом выхода игры(ещё рейтинг не присваивался и оценки не собирались), а частично с нехваткой данных в источниках информации. Эти пропуски не помешают исследованию и их можно оставить.

- В годе релиза данные представлены типом float64,что  связано с наличием пропусков в данных.Заменим пропуски на значение-индикатор 0, чтобы не потерять строки с данными, так как число пропусков более 1,5 %. Года релиза со значением-индикатором не войдут в итоговый срез данных.  Приведём столбец к целочисленному формату,для удобства последующей фильтрации.

In [20]:
df['year_of_release']=df['year_of_release'].fillna(0)

In [21]:
df['year_of_release']=df['year_of_release'].astype('int64')

- Выведем на экран обновлённые значения года релиза.

In [22]:
display(df['year_of_release'].unique())

array([2006, 1985, 2008, 2009, 1996, 1989, 1984, 2005, 1999, 2007, 2010,
       2013, 2004, 1990, 1988, 2002, 2001, 2011, 1998, 2015, 2012, 2014,
       1992, 1997, 1993, 1994, 1982, 2016, 2003, 1986, 2000,    0, 1995,
       1991, 1981, 1987, 1980, 1983])

- Удалим строки с пропусками в столбцах с названием игр и жанром.

In [23]:
df=df.dropna(subset=['name','genre'])

- Выведем обновлённую информацию о датасете.

In [24]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 16954 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   name                16954 non-null  object 
 1   platform            16954 non-null  object 
 2   year_of_release     16954 non-null  int64  
 3   genre               16954 non-null  object 
 4   na_sales            16954 non-null  float64
 5   eu_sales            16948 non-null  float64
 6   jp_sales            16950 non-null  float64
 7   other_region_sales  16954 non-null  float64
 8   critic_score        8242 non-null   float64
 9   user_score          7688 non-null   float64
 10  rating              10085 non-null  object 
dtypes: float64(6), int64(1), object(4)
memory usage: 1.6+ MB


- Заменим пропуски в столбцах с продажами в Европе и Японии средним значением, в зависимости от названия платформы и года выхода игры. Для этого применим метод apply().

In [25]:
def mean_group_eu_sales(row):
    """Функция принимает аргумент row-строка из датафрейма df,
    создаёт переменную group в которую сохраняет отфильтрованный датафрейм по 2 условиям:
    -год релиза совпадает с годом релиза в текущей строке
    -название платформы совпадает с названием платформы в текущей строке.
    В конце функция возвращает среднее значение продаж """
    
    group = df[(df['year_of_release'] == row['year_of_release']) & 
               (df['platform'] == row['platform'])]
    return group['eu_sales'].mean()
df['eu_sales']=df.apply(mean_group_eu_sales,axis=1)

In [26]:
def mean_group_jp_sales(row):
    """Функция принимает аргумент row-строка из датафрейма df,
    создаёт переменную group в которую сохраняет отфильтрованный датафрейм по 2 условиям:
    -год релиза совпадает с годом релиза в текущей строке
    -название платформы совпадает с названием платформы в текущей строке.
    В конце функция возвращает среднее значение продаж """
    
    group = df[(df['year_of_release'] == row['year_of_release']) & 
               (df['platform'] == row['platform'])]
    return group['jp_sales'].mean()
df['jp_sales']=df.apply(mean_group_jp_sales,axis=1)

- Выведем обновлённую информацию о датасете с обработанными пропусками.

In [27]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 16954 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   name                16954 non-null  object 
 1   platform            16954 non-null  object 
 2   year_of_release     16954 non-null  int64  
 3   genre               16954 non-null  object 
 4   na_sales            16954 non-null  float64
 5   eu_sales            16954 non-null  float64
 6   jp_sales            16954 non-null  float64
 7   other_region_sales  16954 non-null  float64
 8   critic_score        8242 non-null   float64
 9   user_score          7688 non-null   float64
 10  rating              10085 non-null  object 
dtypes: float64(6), int64(1), object(4)
memory usage: 1.6+ MB


###  Явные и неявные дубликаты в данных



- Выявим и обработаем неявные дубликаты в категориальных данных. Для этого выведем уникальные значения столбцов с жанром игры,платформы, рейтинга и года релиза. Нормализуем тектовые данные, чтоб убрать неявные дубликаты.

In [28]:
display('Уникальные значения в названии жанра игры')
df['genre'].unique()

'Уникальные значения в названии жанра игры'

array(['Sports', 'Platform', 'Racing', 'Role-Playing', 'Puzzle', 'Misc',
       'Shooter', 'Simulation', 'Action', 'Fighting', 'Adventure',
       'Strategy', 'MISC', 'ROLE-PLAYING', 'RACING', 'ACTION', 'SHOOTER',
       'FIGHTING', 'SPORTS', 'PLATFORM', 'ADVENTURE', 'SIMULATION',
       'PUZZLE', 'STRATEGY'], dtype=object)

In [29]:
display('Уникальные значения в названии платформы')
df['platform'].unique()

'Уникальные значения в названии платформы'

array(['Wii', 'NES', 'GB', 'DS', 'X360', 'PS3', 'PS2', 'SNES', 'GBA',
       'PS4', '3DS', 'N64', 'PS', 'XB', 'PC', '2600', 'PSP', 'XOne',
       'WiiU', 'GC', 'GEN', 'DC', 'PSV', 'SAT', 'SCD', 'WS', 'NG', 'TG16',
       '3DO', 'GG', 'PCFX'], dtype=object)

In [30]:
display('Уникальные значения рейтинга')
df['rating'].unique().tolist()

'Уникальные значения рейтинга'

['E', nan, 'M', 'T', 'E10+', 'K-A', 'AO', 'EC', 'RP']

In [31]:
display('Уникальные значения года релиза')
df['year_of_release'].unique()

'Уникальные значения года релиза'

array([2006, 1985, 2008, 2009, 1996, 1989, 1984, 2005, 1999, 2007, 2010,
       2013, 2004, 1990, 1988, 2002, 2001, 2011, 1998, 2015, 2012, 2014,
       1992, 1997, 1993, 1994, 1982, 2016, 2003, 1986, 2000,    0, 1995,
       1991, 1981, 1987, 1980, 1983])

- Согласно современному рейтингу ESRB рейтинг 'K-A' это старое обозначение современного рейтинга 'E'(for everyone). Поэтому, чтобы убрать неявные дубликаты в столбце с рейтингом, заменим значение 'K-A' на 'E'.

In [32]:
df['rating']=df['rating'].replace('K-A','E')

- Неявные дубликаты встречаются в столбце с названием жанра игры. Приведём его к верхнему регистру. Названия игр тоже приведём к нижнему регистру, чтоб нормализовать текстовые данные. Выводить уникальные названия игр не будем, так как большой объём данных.

In [33]:
df['genre']=df['genre'].str.upper()

In [34]:
df['name']=df['name'].str.lower()

- Проверяем наличие явных дубликатов в данных.

In [35]:
df.duplicated(subset=None,keep='first').sum()

241

- Удаляем явные дубликаты в количестве 241 строки.

In [36]:
df=df.drop_duplicates(subset=None,keep='first',inplace=False)

- В результате процесса очистки данных от явных и неявных дубликатов найдено и удалено 241 строка с явными дубликатами, а текстовые данные с неявными дубликатами приведены к единому регситру, что помогло очистить данные для последующего исседования и категоризации.

- В процессе подготовки данных вы могли что-либо удалять, например строки с пропусками или ошибками, дубликаты и прочее. В этом случае посчитайте количество удалённых строк в абсолютном и относительном значениях.

- В результате очистки данных от пропусков и дубликатов было удалено 2 строки с пропуском в данных в названии игры и жанре, а также 241 строка с явными дубликатами. Посчитаем количество удалённых строк в абсолютном и относительном значениях.

In [37]:
display('Количество удалённых строк в абсолюном значении')
display(start_string_count-df.shape[0])

'Количество удалённых строк в абсолюном значении'

243

In [38]:
display('Количество удалённых строк в относительном значении в %')
display(round((start_string_count-df.shape[0])/df.shape[0]*100,2))

'Количество удалённых строк в относительном значении в %'

1.45

- После проведения предобработки данных напишите общий промежуточный вывод.

- В результате предобработки данных 4 столбца с годом релиза,продажами в Японии и Европе, а также оценками пользователей были приведены к нужному типу. Были убраны неявные дубликаты в столбцах с названием игр и жанра  путём нормализации текстовой информации через приведение к общему регистру. А также удалена 241 строка с явными дубликатами, что поможет использовать очищенные данные для более точного последующего анализа и категоризации.

---

## Фильтрация данных



- Отфильтруем данные,согласно заданию на исследование, за период с 2000 по 2013 год включительно. Сохраним срез данных в новом DataFrame и выведем первые 5 строк.

In [39]:
df_per_2000_to_2013=df[(df['year_of_release']>=2000) & (df['year_of_release']<=2013)]

In [40]:
df_per_2000_to_2013.head()

name platform  year_of_release     genre  na_sales  \
0             wii sports      Wii             2006    SPORTS     41.36   
2         mario kart wii      Wii             2008    RACING     15.68   
3      wii sports resort      Wii             2009    SPORTS     15.61   
6  new super mario bros.       DS             2006  PLATFORM     11.28   
7               wii play      Wii             2006      MISC     13.96   

   eu_sales  jp_sales  other_region_sales  critic_score  user_score rating  
0  0.991364  0.207955                8.45          76.0         8.0      E  
2  0.168556  0.044683                3.29          82.0         8.3      E  
3  0.178489  0.050937                2.95          80.0         8.0      E  
6  0.113088  0.189167                2.88          89.0         8.5      E  
7  0.991364  0.207955                2.84          58.0         6.6      E

---

##  Категоризация данных
    
Проведём категоризацию данных:


- Разделим игры по оценкам пользователей на 3 группы: 'Высокая оценка' (от 8 до 10 включительно),'Средняя оценка' (от 3 до 8, не включая правую границу интервала),'Низкая оценка' (от 0 до 3, не включая правую границу интервала).

In [41]:
import warnings

In [42]:
warnings.simplefilter(action='ignore')

In [43]:
df_per_2000_to_2013['user_score_categories']=pd.cut(df_per_2000_to_2013['user_score'],bins=[0,3,8,10.1],labels=['Низкая оценка','Средняя оценка','Высокая оценка'],right=False)     

- Выведем первые 5 строк датасета

In [44]:
df_per_2000_to_2013.head()

name platform  year_of_release     genre  na_sales  \
0             wii sports      Wii             2006    SPORTS     41.36   
2         mario kart wii      Wii             2008    RACING     15.68   
3      wii sports resort      Wii             2009    SPORTS     15.61   
6  new super mario bros.       DS             2006  PLATFORM     11.28   
7               wii play      Wii             2006      MISC     13.96   

   eu_sales  jp_sales  other_region_sales  critic_score  user_score rating  \
0  0.991364  0.207955                8.45          76.0         8.0      E   
2  0.168556  0.044683                3.29          82.0         8.3      E   
3  0.178489  0.050937                2.95          80.0         8.0      E   
6  0.113088  0.189167                2.88          89.0         8.5      E   
7  0.991364  0.207955                2.84          58.0         6.6      E   

  user_score_categories  
0        Высокая оценка  
2        Высокая оценка  
3        Высокая оценка  
6        Высокая оценка  
7        Средняя оценка

- Разделим  игры по оценкам критиков и выделим следующие категории: высокая оценка (от 80 до 100 включительно), средняя оценка (от 30 до 80, не включая правую границу интервала) и низкая оценка (от 0 до 30, не включая правую границу интервала).

In [45]:
df_per_2000_to_2013['critic_score_categories']=pd.cut(df_per_2000_to_2013['critic_score'],bins=[0,30,80,101],labels=['Низкая оценка','Средняя оценка','Высокая оценка'],right=False)     

- Выведем первые 5 строк датасета.

In [46]:
df_per_2000_to_2013.head()

name platform  year_of_release     genre  na_sales  \
0             wii sports      Wii             2006    SPORTS     41.36   
2         mario kart wii      Wii             2008    RACING     15.68   
3      wii sports resort      Wii             2009    SPORTS     15.61   
6  new super mario bros.       DS             2006  PLATFORM     11.28   
7               wii play      Wii             2006      MISC     13.96   

   eu_sales  jp_sales  other_region_sales  critic_score  user_score rating  \
0  0.991364  0.207955                8.45          76.0         8.0      E   
2  0.168556  0.044683                3.29          82.0         8.3      E   
3  0.178489  0.050937                2.95          80.0         8.0      E   
6  0.113088  0.189167                2.88          89.0         8.5      E   
7  0.991364  0.207955                2.84          58.0         6.6      E   

  user_score_categories critic_score_categories  
0        Высокая оценка          Средняя оценка  
2        Высокая оценка          Высокая оценка  
3        Высокая оценка          Высокая оценка  
6        Высокая оценка          Высокая оценка  
7        Средняя оценка          Средняя оценка

- Сгруппируем данные по по категориям оценок пользователей и посчитаем количество игр в каждой категории.

In [47]:
display(df_per_2000_to_2013.groupby('user_score_categories').agg({'name':'count'}))

name
user_score_categories      
Низкая оценка           116
Средняя оценка         4081
Высокая оценка         2286

- Сгруппируем данные по категориям оценок критиков и посчитаем количество игр в каждой категории.

In [48]:
display(df_per_2000_to_2013.groupby('critic_score_categories').agg({'name':'count'}))

name
critic_score_categories      
Низкая оценка              55
Средняя оценка           5422
Высокая оценка           1692

- Выделим топ-7 платформ по количеству игр, выпущенных за весь актуальный период.

In [49]:
df_per_2000_to_2013.platform.value_counts().head(7)

PS2     2127
DS      2120
Wii     1275
PSP     1180
X360    1121
PS3     1087
GBA      811
Name: platform, dtype: int64

---

##  Итоговый вывод


- Для исследование использовался датасет с информации о выпущенных играх,платформах, годах релиза и их оценками, а также рейтингом. Объёмом датасета составлял 16956 строк и 11 столбцов. В результате ознакомления с данными были выявлены несоответствия данных в отдельных столбцах их типу,а также пропуски значений в 6 столбцах. Данные в столбцах с годом релиза, продажами в Европе и Японии, а также оценками пользователей были приведены к верному типу. В столбцах с названиеями игр и жанрами было 2 общих пропуска и строки с ними были удалены. Пропуски в столбце с годом релиза были заменены на значение-индикатор 0, который не сказался на дальнейшем анализе. Пропуски в столбцах с количеством продаж в Европе и Японии были заменены среднем значением с учётом платформы и года релиза. Пропуски в столбцах с оценками критиков и пользователей, а также рейтингом остались без изменений. После очистки данных для исследования был сделан срез за период с 2000 по 2013 года включительно. Данные были категоризированы по оценкам критиков и пользователей на 3 группы:'Низкая оценка','Средняя оценка' и 'Высокая оценка'. В результате было выявлено, что наибольшее количество игр в этом периоде имеют среднюю и высокую оценки, что указывает на высокий уровень выпущенных игр. Также были выделены топ-7 платформ по количеству выпущенных игр, самые популярные из них: 'PS2','DS' и 'Wii'.